test/test-clusters/src/main/java/org/elasticsearch/test/cluster/util/resource/FileResource.java

```diff
@@ -14,18 +14,19 @@
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.StandardOpenOption;
+import java.util.function.Supplier;
 
 public class FileResource implements Resource {
-    private final Path file;
+    private final Supplier<Path> file;
 
-    FileResource(Path file) {
+    FileResource(Supplier<Path> file) {
```

Comment:
We need an option to defer that file reference as it is created when the krb5kdc container has been started and is not available before.